### ISSUE TITLE
early stopping in training

### ISSUE THREAD

**Issue by vishnukv64** [ultralytics/yolov5#294]
Hey, 
I just wanted to know how to automatically stop the training if the loss doesnt decrease for say 10 epochs and save the best and last weights?

if there is snothing i could do with the results.txt or something

Thanks

**Comment by NanoCode012** [comment]
Hm, if you run it **without** `--nosave`, you can just cntrl+c the run. Then call ,
https://github.com/ultralytics/yolov5/blob/5e2429e6188254ad69401b172be3772efb9698e1/utils/utils.py#L627-L633

Put below in yolov5 folder, then call via `python <filename>`
```python
from utils.utils import strip_optimizer;

strip_optimizer('weights/best.pt')
```

**Comment by glenn-jocher** [comment]
https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.ReduceLROnPlateau might also be of interest.
